python-frontend/src/main/java/org/sonar/python/semantic/v2/SymbolsModuleTypeProvider.java

```diff
@@ -140,18 +141,16 @@ private PythonType convertToFunctionType(FunctionSymbol symbol, Map<Symbol, Pyth
         .withHasVariadicParameter(symbol.hasVariadicParameter())
         .withDefinitionLocation(symbol.definitionLocation());
     FunctionType functionType = functionTypeBuilder.build();
-    if (returnType instanceof LazyType lazyType) {
-      lazyType.addConsumer(functionType::resolveLazyReturnType);
-    }
     createdTypesBySymbol.put(symbol, functionType);
     return functionType;
   }
 
-  private PythonType getReturnTypeFromSymbol(FunctionSymbol symbol) {
+  private TypeWrapper getReturnTypeFromSymbol(FunctionSymbol symbol) {
     var returnTypeFqns = getReturnTypeFqn(symbol);
     var returnTypeList = returnTypeFqns.stream().map(lazyTypesContext::getOrCreateLazyTypeWrapper).map(ObjectType::new).toList();
     //TODO Support type unions (SONARPY-2132)
-    return returnTypeList.size() == 1 ? returnTypeList.get(0) : PythonType.UNKNOWN;
+    var type = returnTypeList.size() == 1 ? returnTypeList.get(0) : PythonType.UNKNOWN;
+    return new SimpleTypeWrapper(type);
```

Comment:
At first reading, I thought creating a SimpleTypeWrapper was wrong since there is a LazyType involved. It's fine now that ObjectType has a TypeWrapper inside now.